### ISSUE TITLE
Search modal does not constrain tabbing within the modal.

### ISSUE THREAD

**Issue by katiebethbrown** [Automattic/jetpack#23396]
### Impacted plugin
Jetpack

### Issue
It is possible for a keyboard user to tab out of the search modal without closing it. Once this happens, there is no visual reference for the current location of focus, and it may be very difficult to return to the modal.

While the modal is open, keyboard focus should be limited to only allow access to focusable elements that are inside the modal.

See WCAG 2.4.3: https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html

### Other Information
This issue was surfaced to team 51 by an a11y contractor working on a partner site.

### Additional Note
I'm not sure whether I'm replicating the original issue or not. I see focus shift from a search result to the browser back button, when I would have expected tab to take me through results:

![CleanShot 2022-03-11 at 14 21 31](https://user-images.githubusercontent.com/42279181/157971271-4257f3cc-af48-48b5-b75f-18606e9941c7.gif)
